src/DocumentationBrowserViewExtension/DocumentationBrowserViewModel.cs

```diff
@@ -199,40 +208,34 @@ internal string GetContent()
             return this.content;
         }
 
-        /// <summary>
-        /// Updates the content to be displayed in the browser and triggers the LinkChanged action.
-        /// </summary>
-        /// <param name="newContent">The content to display.</param>
-        internal void UpdateContent(string newContent)
-        {
-            if (string.IsNullOrWhiteSpace(newContent))
-                return;
-
-            this.content = newContent;
-            this.isRemoteResource = false;
-            OnLinkChanged(null);
-        }
-
-        private void LoadContentFromFile(Uri link)
-        {
-            var path = ResolveFilePath(link);
-            this.content = File.ReadAllText(path);
-
-            // when we have no content fall back to the no content page
-            if (string.IsNullOrWhiteSpace(this.content))
-                throw new TargetException();
-        }
-
         private string LoadContentFromResources(string name)
         {
             if (string.IsNullOrWhiteSpace(name))
                 throw new ArgumentNullException(nameof(name));
+            
             string result;
+            // If an assembly was specified in the uri, the resource will be searched there.
+            Assembly assembly;
+            var assemblyIndex = name.LastIndexOf(";");
+            if (assemblyIndex != -1)
+            {
+                var assemblyName = name.Substring(0, assemblyIndex);
+                // Ignore version and public key, in case they were specified.
+                var versionIndex = assemblyName.IndexOf(";");
+                if (versionIndex != -1)
+                {
+                    assemblyName = assemblyName.Substring(0, versionIndex);
+                }
+                assembly = Assembly.Load(assemblyName);
```

Comment:
hmm, I think we should avoid adding more locations where we load assemblies...
If this is a zero touch node, the assembly should already be loaded, what about if limit this to searching the loaded assemblies in the current app domain - this will require that node authors add their documentation to the same assembly as their nodes.

@aparajit-pratap FYI.

Reply:
You are right. I think I tried doing it that way but I found some issues with it. Will try that again.